zephyr related changelog update (#2828)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

.. NOTE TO MAINTAINERS: The latest Zephyr commit appears in multiple places; make sure you update them all.

> The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repository up to and including ``0aaee6dcf5``, plus some |NCS| specific additions.

Author: Checked, Yes it is correct.

  * Modified I2C1 and SPI2 pin assignments in the :ref:`zephyr:nrf5340pdk_nrf5340` to match
    the standard location for I2C and SPI in the Arduino header.
  * Added nRF52820 nrfx defines for emulation on the :ref:`zephyr:nrf52833dk_nrf52833`.
  * Support for nRF21540 Devkit.
  

* Networking:

  * Switched networking threads to use kernel stack.

  * LwM2M:

    * Fixed a bug where a FOTA socket was not closed after the download (PULL mode).
    * Added a Kconfig option :option:`CONFIG_LWM2M_SECONDS_TO_UPDATE_EARLY` that specifies how long before the time-out the Registration Update will be sent.
    * Added ObjLnk resource type support.
    * Fixed Security and Server object instance matching.
    * Fix poll fds handling (in case there is another socket open).
    * Made send() calls on the same socket thread-safe.

>     * Fixed insufficient `sockaddr` structure size provided on IPv6 socket while IPv4 enabled as well.

Reviewer: I'd add two more entries here:
```
    * Fixed PUSH mode FOTA.
    * Fixed bootstrap procedure.
```

    * The default ``clean_session`` value is now configurable with Kconfig (see :option:`CONFIG_MQTT_CLEAN_SESSION`).
    * Prevented double CONNACK event notification on server reject.

  * OpenThread:

    * Updated the OpenThread revision to upstream commit ``ac86fe52e62e60a66aeeb1c905cb1294709147e9``.
    * Implemented a missing ``enable`` API function for the OpenThread interface.
    * Cleaned up the OpenThread Kconfig file.
      OpenThread dependencies are now enabled automatically.
    * Allowed the application to register a callback function for OpenThread state changes.
    * Reimplemented the logger glue layer for better performance.
    * Updated the OpenThread thread priority class to be configurable.
    * Added several Kconfig options to customize the OpenThread stack.
    * Added Sleep to Transmit as hardware radio capability (``OT_RADIO_CAPS_SLEEP_TO_TX``).
    * Removed retransmissions from the radio capabilities (``OT_RADIO_CAPS_TRANSMIT_RETRIES``).
    * Added a configuration option to select the OpenThread version (either 1.1 or 1.2).
    * Added configuration options for NCP vendor hooks (see :option:`CONFIG_OPENTHREAD_NCP_VENDOR_HOOK_SOURCE`).

>     * Allowed use of custom mbed TLS (see :option:`CONFIG_OPENTHREAD_MBEDTLS_LIB_NAME`).

Author: will be replaced by CONFIG_OPENTHREAD_MBEDTLS_TARGET in upcoming upmerge

    * Removed double-buffering in UART send.
    * Fixed the network initialization when :option:`CONFIG_NET_CONFIG_MY_IPV6_ADDR` is not set.
    * Added a Kconfig option that allows to link Zephyr with precompiled OpenThread libraries (see :option:`CONFIG_OPENTHREAD_SOURCES`).
    * Added a Kconfig option to compile with Diagnostic functions support (see :option:`CONFIG_OPENTHREAD_DIAG`).

  * Socket offloading:

    * Removed dependency to the :option:`CONFIG_NET_SOCKETS_POSIX_NAMES` configuration option.
    * Fixed an issue where the network interface was missing for offloaded drivers (#27037).
    * Updated the ``close()`` socket call to no longer use ``ioctl()`` underneath.
      It now has a separate entry in a socket vtable.

  * IP:

    * Fixed an issue where IPv6 RS messages did not comply with RFC4291.

>     * Added infrastructure for collecting stack timing statistics for network packet pass-through (see :option:`CONFIG_NET_PKT_TXTIME_STATS_DETAIL` and :option:`CONFIG_NET_PKT_RXTIME_STATS_DETAIL`).

Author: record is technically correct, It also references Kconfig option a user can look into.

    * Reimplemented the logger glue layer for better performance.
    * Updated the OpenThread thread priority class to be configurable.
    * Added several Kconfig options to customize the OpenThread stack.

>     * Updated the OpenThread revision to upstream commit ``ac86fe52e62e60a66aeeb1c905cb1294709147e9``

Reviewer: I'd just update the existing entry with current SHA (line 89).

>     * radio: Added sleep to tx as hw radio capability.

Reviewer: Let's mention the capability name here (`OT_RADIO_CAPS_SLEEP_TO_TX`).

    * net: Added config for OpenThread version selection (either 1.1 or 1.2).

>     * radio: Removed retransmissions from radio capability.

Reviewer: `OT_RADIO_CAPS_TRANSMIT_RETRIES`

    * Added config options for NCP vendor hooks, see :option:`CONMFIG_OPENTHREAD_NCP_VENDOR_HOOK_SOURCE`.

>     * Allow use of custom mbedtls

Reviewer: let's mention the option here: ":option:\`CONFIG_OPENTHREAD_MBEDTLS_TARGET\`"

    * platform: Removed double-buffering in UART send.
    * Fix the network initialization when :option:`CONFIG_NET_CONFIG_MY_IPV6_ADDR` is not set

>     * Allow to build Zephyrs OpenThread port from user's custom sources, see :option:`CONFIG_OPENTHREAD_SOURCES`.

Reviewer: The meaning of this change was slightly different:
```
Added a Kconfig option that allows to link Zephyr with pre-compiled OpenThread libraries (:option:`CONFIG_OPENTHREAD_IMPLEMENTATION`).
```

    * Allow to compile Diagnostic functions support, see :option:`CONFIG_OPENTHREAD_DIA`.

  * Socket offloading:

    * Removed dependency to the :option:`CONFIG_NET_SOCKETS_POSIX_NAMES` configuration option.

>     * Fixed Network interface was missing for offloaded drivers (#27037).

Reviewer: I'd also add the following change:
```
* ``close()`` socket call is no longer using ``ioctl()`` underneath, it has a separate entry in a socket vtable.
```

>   * PPP:

Reviewer: I don't think we need to mention PPP changes here, it's not something we use with Nordic HW.

    * Added initial support for PAP authentication.
    * Automatic select CONFIG_RING_BUFFER as it is required.
    * Refactored option negotiation.
    * Fixed a race condition when ppp_fsm_open() was called in CLOSED state.
    * Added remove network address in ip{,v6}cp_down implementations which allow to recognize link-down by the application.
    * Fsm: use packet work object instead of shared one, which allow to handle more than one packet at one time.

>   * IP:

Reviewer: Missing empty line.

    * Fixed: IPv6 RS message doesn't comply RFC4291.
    * Added network packet pass-through stack timing statistics infrastructure.
      see :option:`CONFIG_NET_PKT_TXTIME_STATS_DETAI`, :option:`CONFIG_NET_PKT_RXTIME_STATS_DETAI`.

>   * TCP:

Reviewer: Missing empty line.

    Use :cpp:func:`bt_le_adv_start` instead, with :cpp:member:`bt_le_adv_param::peer` set to the remote peer's address.
  * Deprecated the ``BT_LE_ADV_*`` macros.
    Use the ``BT_GAP_ADV_*`` enums instead.
  * Updated L2CAP RX MTU to be controlled by :option:`CONFIG_BT_L2CAP_RX_MTU` (instead of :option:`CONFIG_BT_RX_BUF_LEN`) when :option:`CONFIG_BT_HCI_ACL_FLOW_CONTROL` is disabled.
    If :option:`CONFIG_BT_RX_BUF_LEN` is changed from its default value, :option:`CONFIG_BT_L2CAP_RX_MTU` should be set to ``CONFIG_BT_RX_BUF_LEN - 8``.
  * Added support for periodic advertisement to the Host.
  * Added a :cpp:member:`bt_conn_auth_cb::bond_deleted` callback to the Host.

>   * Allowed to start a persistent advertiser when the maximum number of connections is reached.

Reviewer 2: No, we want to use the same style for all items and say what was changed.
So if "Allowed to... start" is not correct, maybe "Added support for starting ..." or "Added an option to start ..."?

>   * Allowed to start a persistent advertiser when the maximum number of connections is reached.

Author: @joerchan  Is it OK now?

    Use :cpp:func:`bt_le_adv_start` instead, with :cpp:member:`bt_le_adv_param::peer` set to the remote peer's address.
  * Deprecated the ``BT_LE_ADV_*`` macros.
    Use the ``BT_GAP_ADV_*`` enums instead.
  * L2CAP RX MTU is now controlled by CONFIG_BT_L2CAP_RX_MTU when
    CONFIG_BT_ACL_FLOW_CONTROL is disabled, previously this was controlled
    by CONFIG_BT_RX_BUF_LEN. If CONFIG_BT_RX_BUF_LEN has been changed from its
    default value then CONFIG_BT_L2CAP_RX_MTU should be set to
    CONFIG_BT_RX_BUF_LEN - 8.
  * host: Added support for periodic advertisement.
  * host: Added bond_deleted callback.
  * host: Refactor the BR/EDR SSP feature out into it's own source file.
  * host: Allow Start persistent advertiser with maximum connection number reached.
  * host: Fixed settings of Advertisng Data on extended advertising instances.
  * host: SMP: Reject legacy pairing early in SC only mode.

>   * Fixed infinite recursion possibility in persistent advertising.

Reviewer 3: This issue never existed downstream. We are pulling in both the regression, and the fix at the same time.
```suggestion

```

  * GATT: Fixed issue with bt_gatt_service_unregister not clearing CCC information, which might cause no space to stor ccc configuration.
  * L2CAP: Elevate the security level before sending the connection request if the application has set a required security level on the channel.
  * GATT: Added option to disable security checks, see :option: `CONFIG_BT_CONN_DISABLE_SECURITY`.
  * GATT: Automatic discover of CCC when subscribing, see :option:`CONFIG_BT_GATT_AUTO_DISCOVER_CCC`.
  * GATT: Fixed the issue where peripheral may not store CCC in non-volatile memory in case of multiple CCC change (due to race condition).

> 

Reviewer 3: host: Fixed deadlock in receiving disconnected event when disconnecting with pending GATT Write Commands
host: Fixed issue where persistent advertiser would not be started due to  race-condition

Author: added

  * Updated the Controller to only use control procedures supported by the peer.
  * Added support for the nRF52820 SoC.
  * Removed the legacy Controller.
  * Implemented a function to remove auxiliary advertising sets (``ll_adv_aux_set_remove()``).
  * Implemented a function to remove all primary channels and auxiliary channels of an advertising set (``ll_adv_aux_set_clear()``).
  * Fixed overflow that could happen when using uninitialized PDU.
  * Removed redundant :option:`CONFIG_BT_LL_SW_SPLIT` conditional.

>   * Enforced support for Read RSSI command if the Connection State is supported.

Reviewer 2: Is "enforced support" correct here? Or should it be "added support", or "Enforced that the Read RSSI command is used if..."/"Added code to enforce that ..."?

Author: `Enforced that the Read RSSI command is supported if the Connection State is supported.`
This is BLE standard requirement.

  * Added missing aux acquire on periodic advertising.
  * Updated the implementation to schedule non-overlapping sync PDUs.
  * Fixed the handling of HCI commands for extended advertising.
  * Added a terminate event for extended advertising.
  * Filled the missing Periodic Advertising interval in the Extended Advertising Report when auxiliary PDUs contain Sync Info fields.
  * Filled the referenced event counter of the Periodic Advertising SYNC_IND PDU into the Sync Info structure in the Common Extended Advertising Header Format.
  * Switched HCI threads to use the kernel stack.

* Bluetooth Mesh:

  * Removed the ``net_idx`` parameter from the Health Client model APIs because it can be derived (by the stack) from the ``app_idx`` parameter.
  * Documented :ref:`Mesh Shell commands <zephyr:bluetooth_mesh_shell>`.
  * Allowed to configure the advertiser stack size (see :option:`CONFIG_BT_MESH_ADV_STACK_SIZE`).
  * Resolved a corner case where the segmented sending would be rescheduled before the segments were done sending.

>   * Fixed TX seg buffer with NULL pointer reference if enabled mesh-friendly.

Reviewer 2: I don't understand what we're trying to say here.

Author: `mesh-friendly` is a BLE mesh concept, I beat `TX seg` is a well.
@trond-snekvik How to rewrite this?

Reviewer 4: ```suggestion
  * Fixed dangling transport segmentation buffer pointer when Friend feature is enabled
```

  * Switched advertising threads to use the kernel stack.

* Bluetooth shell:

>   * Added an advertising option for undirected one-time advertising, directed advertising using identity, and peer direct is RPA option.

Reviewer 2: What does "is RPA option" mean here?

Reviewer 3: "peer direct is RPA" option.

I think a better phrasing is this "Directed advertising to privacy-enabled peer" option.

This sentence might be better to split across 3 lines, since it is adding support for 3 independant options:

```suggestion
  * Added an advertising option for undirected one-time advertising
  * Added an advertising option for advertising using identity address when local privacy is enabled.
  * Added an advertising option for directed advertising to privacy-enabled peer when local privacy is disabled.
```

  * Updated the info command to print PHY and data length information.

* Drivers:

  * Bluetooth HCI:

>     * Added missing ``gpio_dt_flags`` to :cpp:class:`spi_cs_control` in the SPI driver (spi_hci).

Reviewer 2: Is "spi_cs_control" instead of "spi_cs_config" correct?
And what does "spi_hci" refer to?

Author: "spi_cs_control"  is correct.
changed to `SPI HCI` which is used in Kconfig prompt.
refer to HCI SPI driver implemented here: zephyr/drivers/bluetooth/hci/spi.c

  * Updated the implementation to schedule non-overlapping sync PDUs.
  * Fixed the handling of HCI commands for extended advertising.
  * Added a terminate event for extended advertising.
  * Filled the missing Periodic Advertising interval in the Extended Advertising Report when auxiliary PDUs contain Sync Info fields.
  * Filled the referenced event counter of the Periodic Advertising SYNC_IND PDU into the Sync Info structure in the Common Extended Advertising Header Format.
  * Switched HCI threads to use the kernel stack.

* Bluetooth Mesh:

  * Removed the ``net_idx`` parameter from the Health Client model APIs because it can be derived (by the stack) from the ``app_idx`` parameter.
  * Documented :ref:`Mesh Shell commands <zephyr:bluetooth_mesh_shell>`.
  * Allowed to configure the advertiser stack size (see :option:`CONFIG_BT_MESH_ADV_STACK_SIZE`).
  * Resolved a corner case where the segmented sending would be rescheduled before the segments were done sending.
  * Fixed dangling transport segmentation buffer pointer when Friend feature is enabled.
  * Switched advertising threads to use the kernel stack.

* Bluetooth shell:

>   * Added an advertising option for undirected one-time advertising

Reviewer 2: ```suggestion
  * Added an advertising option for undirected one-time advertising.
```

    * Added support for sub-word lengths of read and write transfers in the nRF QSPI NOR flash driver (nrf_qspi_nor).
    * Improved the handling of erase operations in the nRF QSPI NOR flash driver (nrf_qspi_nor), the AT45 family flash driver (spi_flash_at45), and the SPI NOR flash driver (spi_nor).
      Now the operation is not started if it cannot be completed successfully.
    * Established the unrestricted alignment of flash reads for all drivers.
    * Enhanced the nRF QSPI NOR flash driver (nrf_qspi_nor) so that it supports unaligned read offset, read length, and buffer offset.
    * Added SFDP support in the SPI NOR flash driver (spi_nor).
    * Fixed a regression in the nRF flash driver (soc_flash_nrf) when using the :option:`CONFIG_BT_CTLR_LOW_LAT` option.

  * EEPROM:

>     * Fixed at2x cs gpio flags extraction from DT.

Reviewer 2: I don't understand what we're trying to say here.

Author: Rephrased and moved above.

    * Changed the nRF UART nrfx drivers (uart_nrfx_uart/uarte) to use the DT ``hw-flow-control`` property instead of Kconfig options.
    * Fixed disabling of the TX interrupt in the uart_nrfx_uart driver.
    * Fixed the uart_nrfx_uarte driver to prevent spurious :cpp:enumerator:`UART_RX_BUF_REQUEST <uart_interface::UART_RX_BUF_REQUEST>` events.

>     * Removed counters reset from :cpp:func:`uart_rx_enable` in the nrf_uarte driver.

Reviewer 2: Is "uart_rx_enable" correct?

Author: yes, https://docs.zephyrproject.org/latest/reference/peripherals/uart.html?highlight=uart_rx_enable#c.uart_rx_enable

    * Introduced names for threads used in USB classes, to aid debugging.
    * The usb_enable() function, which, for some samples, was invoked
      automatically on system boot up, now needs to be explicitly called
      by the application in order to enable the USB subsystem. If your
      application relies on any of the following Kconfig options, then
      it shall also enable the USB subsystem:
      * :option:`CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_ACM`
      * :option:`CONFIG_USB_DEVICE_NETWORK_ECM`
      * :option:`CONFIG_USB_DEVICE_NETWORK_EEM`
      * :option:`CONFIG_USB_DEVICE_NETWORK_RNDIS`
      * :option:`CONFIG_TRACING_BACKEND_USB`
      * :option:`CONFIG_USB_UART_CONSOLE`
    * cdc class: Do not duplicate USB transfer slots for OUT transfers.
    * cdc_acm: re-trigger transfer for OUT endpoint on resume. 
    * nRF: Fix for remote wake up request.
    * hid: Allow to send data only in CONFIGURED state.
    * Switched to using kernel stack.
    * audio: Correct typo in USB audio DTS bindings.

  * Watchdog:

    * Updated the description of the :cpp:func:`wdt_feed` API function to reflect an additional error return code.

>   * IEEE 802.15.4:

Reviewer 5: This should be moved up, to keep the alphabetical order.

Author: moved

    * Unified endpoint helper macros across all USB device drivers.
    * Fixed handling of fragmented transfers on the control OUT endpoint in the Nordic Semiconductor USB Device Controller driver (usb_dc_nrfx).
    * Introduced names for threads used in USB classes, to aid debugging.
    * Updated the way the :cpp:func:`usb_enable` function should be used.
      For some samples, this function was invoked automatically on system boot-up to enable the USB subsystem, but now it must be called explicitly by the application.
      If your application relies on any of the following Kconfig options, it must also enable the USB subsystem:

      * :option:`CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_ACM`
      * :option:`CONFIG_USB_DEVICE_NETWORK_ECM`
      * :option:`CONFIG_USB_DEVICE_NETWORK_EEM`
      * :option:`CONFIG_USB_DEVICE_NETWORK_RNDIS`
      * :option:`CONFIG_TRACING_BACKEND_USB`
      * :option:`CONFIG_USB_UART_CONSOLE`

    * Updated the implementation of the CDC class to not duplicate USB transfer slots for OUT transfers.

>     * Updated the implementation of CDC ACM to re-trigger transfer for the OUT endpoint on resume.

Reviewer 6: The first one is fixing an issue introduced by the second one. If we need to say about all bug fixes then please concatenate those two into:
`Fix an issue that CDC ACM was not accepting OUT transfers after Resume from Suspend.`

    * Fixed an issue with remote wake-up requests in the nRF driver.
    * Updated the implementation of the HID class to allow sending data only in CONFIGURED state.

>     * Switched to using the kernel stack.

Reviewer 6: This one says nothing to me and remind me nothing. Could You please point me to the commit that You base this entry on?

Author: As you wish: https://github.com/zephyrproject-rtos/zephyr/pull/27308

Reviewer 6: ``` diff
-     * Switched to using the kernel stack.
+     * Use kernel stack for threads not running in user space.
```

Reviewer 2: ```suggestion
    * Updated to use the kernel stack for thread not running in user space.
```

>     * Corrected a typo in USB audio DTS bindings.

Reviewer 6: This is not worth mentioning.

    The functions will now return an error code in this case.
  * Fixed a garbage-collection issue in the NVS subsystem.
  * Added the Kconfig option :option:`CONFIG_FS_FATFS_EXFAT` for enabling exFAT support.
  * Added support for file open flags to fs and POSIX API.

* Management:

  * MCUmgr:

    * Moved mcumgr into its own directory.
    * Switched UDP port to use the kernel stack.
    * Added missing socket close in error path for SMP.

  * Added support for Open Supervised Device Protocol (OSDP) (see :option:`CONFIG_OSDP`).

  * updatehub:

    * Moved updatehub from :file:`lib` to :file:`subsys/mgmt`.
    * Fixed out-of-bounds access and added a return value check for ``flash_img_init()``.
    * Fixed a ``getaddrinfo`` resource leak.

* Settings:

  * Updated the implementation to return an error rather than faulting if there is an attempt to read a setting from a channel that does not support reading.
  * Disallowed modifying the content of a static subtree name.

* LVGL:

  * Updated the library to the new major release v7.0.2.
  * Aligned LVGL Kconfig constants with suggested defaults from upstream.

* Tracing:

>   * Updated the API to check if the init function exists prior to calling it.

Reviewer 2: I changed "tracing backed API" to just "API" here - is that okay?

Reviewer 6: I think I've added this. Fair enough "API" seems better.

* Logging:

  * Fixed immediate logging with multiple backends.
  * Switched logging thread to use the kernel stack.
  * Allowed users to disable all shell backends at once using :option:`CONFIG_SHELL_LOG_BACKEND`.
  * Added a logging backend for the Spinel protocol.
  * Fixed timestamp calculation when using NEWLIB.

* Shell:

  * Switched to use the kernel stack.
  * Fixed the select command.
  * Fixed prompting dynamic commands.

* libc:

  * Simplified newlib malloc arena definition.

* Devicetree:

  * Removed all nRF-specific aliases to particular hardware peripherals, because they are no longer needed now that nodes can be addressed by node labels.
    For example, you should now use ``DT_NODELABEL(i2c0)`` instead of ``DT_ALIAS(i2c_0)``.

>     If you use something similar to ``source $(SOC_DIR)/<path>``, change it to ``rsource <relative>/<path>`` or similar.

Reviewer 2: Is this a continuation of the first item in this list, or should it be a separate item?

Author: should go after `* BOARD, SOC, DTS, and ARCH roots can now be specified in each module's :file:`zephyr/module.yml` file (see :ref:`modules_build_settings`).` Patch introduced by https://github.com/zephyrproject-rtos/zephyr/pull/26715 + https://github.com/zephyrproject-rtos/zephyr/pull/27431/files
I moved to `build system ` as more relevant.

  * Added a number of iterable section macros to the set of linker macros, including ``Z_ITERABLE_SECTION_ROM`` and ``Z_ITERABLE_SECTION_RAM``.
  * Added a new Zephyr Build Configuration package with support for specific build configuration for Zephyr derivatives (including forks).
    See :ref:`zephyr:cmake_pkg` for more information.
  * Removed the set of ``*_if_kconfig()`` CMake functions.
    Use ``_ifdef(CONFIG_ ...)`` instead.

* Samples:

  * Updated the :ref:`zephyr:nrf-system-off-sample` to better support low-power states of Nordic Semiconductor devices.
  * Updated the :ref:`zephyr:usb_mass` to perform all application-level configuration before the USB subsystem starts.
    The sample now also supports FAT file systems on external storage.
  * Updated the :ref:`zephyr:nvs-sample` sample to do a full chip erase when flashing.
  * Fixed the build of the :ref:`zephyr:bluetooth-mesh-onoff-level-lighting-vnd-sample` application with mcumgr.
  * Added new commands ``write_unaligned`` and ``write_pattern`` to the :ref:`zephyr:samples_flash_shell`.
  * Fixed the ``cmd_hdr`` and ``acl_hdr`` usage in the :ref:`zephyr:bluetooth-hci-spi-sample` sample.
  * Removed the NFC sample.

>   * Fixed an uninitialized ``spi_cfg`` in the :ref:`zephyr:i2c_fujitsu_fram` sample.

Reviewer 2: Is this the correct sample? And do we need really need this line in the highlights for NCS?

Author: removed

  * Updated the configuration for extended advertising in the :ref:`zephyr:bluetooth-hci-uart-sample` and :ref:`zephyr:bluetooth-hci-rpmsg-sample` samples.

* Modules:

  * Introduced a ``depends`` keyword that can be added to a module's :file:`module.yml` file to declare dependencies to other modules.
    This allows to correctly establish the order of processing.

* Testing infrastructure:

  * sanitycheck:

    * Added an ``integration`` option that is used to list platforms to use in integration testing (CI) and avoids whitelisting platforms.

>     * Updated to not expect results from build_only instances.

Reviewer 2: Is it sanitycheck that should not expect results, or is it the user that should not expect results from sanitycheck?

Author: sanitycheck - @PerMac Can you confirm?

Reviewer 7: TL,DR it is sanitycheck that should not expect results
This is more of an internal change/fix in sanitycheck script performance. I'm not sure if it has any consequences for users. It changes the way how statuses of tests are handled inside the script. The tests which were only built don't have 'PASS' status internally during the script execution, which status is obtained after the test was executed on a platform and passed. This PR makes that sanitycheck is not expecting 'PASS' status from build_only instances. Because of this if the build_only instance is tested it still gets a pass in the end in the XML report.
